### PR TITLE
Adds signals for gaining and losing traits

### PR DESCRIPTION
**Pull request by Rohesie**
Atomizing #48923
That way we can react on gained and lost traits, pretty self-explanatory.

### CODE REVIEW

File: code/__DEFINES/dcs/signals.dm [hunk 1]
```diff
@@ -40,6 +40,9 @@
 #define COMSIG_PARENT_QDELETING "parent_qdeleting"
 /// generic topic handler (usr, href_list)
 #define COMSIG_TOPIC "handle_topic"
+/// Trait signals
+#define COMPONENT_ADD_TRAIT (1<<0)
```
Comment (Qustinnus) [inline code/__DEFINES/dcs/signals.dm]: why this define name? its not a component

Comment (spookydonut) [reply code/__DEFINES/dcs/signals.dm]: using numbers for signal values is not a great idea

Comment (spookydonut) [reply code/__DEFINES/dcs/signals.dm]: should likely just have SIGNAL_TRAIT_REMOVED(trait) and SIGNAL_TRAIT_ADDED(trait)

Comment (Rohesie) [reply code/__DEFINES/dcs/signals.dm]: `COMPONENT_(something)` is how we tend to define the signal-based flags.
Can do the change spooky suggests, though.

Comment (Qustinnus) [reply code/__DEFINES/dcs/signals.dm]: but this isnt a flag its a signal?

**Pull request merged**